cedar-policy-validator/src/typecheck.rs

```diff
@@ -1149,6 +1180,7 @@ impl<'a> Typechecker<'a> {
                     expr,
                     Type::any_entity_reference(),
                     type_errors,
+                    |_| Some(UnexpectedTypeHelp::TypeTestNotSupported),
```

Comment:
this is the case where we have something like `principal.manager is User::"alice"`?  should we suggest using `==` instead of `is` to compare two entity references?

Reply:
That's already suggested by the parser when the cst to ast conversion fails

Reply:
Hmm I see.  When do we actually hit this case then?

Reply:
Something like `1 is Long`, if `Long` were a valid entity type, so not often

Reply:
The error also makes sense if some did `1 is User` expect it to be `false` because `1` isn't a user.

Reply:
Makes sense.